src/co/edu/konradlorenz/model/TarjetaDebito.java

```diff
@@ -0,0 +1,54 @@
+package co.edu.konradlorenz.model;
+
+public class TarjetaDebito implements Tarjeta{
```

Comment:
Hay un problema grande con la Interface y esta clase.
sugiero que los atributos (numeroTarjeta y pin) lo tenga la clase Cliente, así a Cliente se le implementa la interface Tarjeta con solo el método esTarjetaValida(), porque el resto de métodos de la interface Tarjeta son realmente get and set de atributos propios de TarjetaDebito, que ahora irían en Cliente.
Cabe recalcar que pin sí lleva get y set, porque si tu tarjeta es valida puedes cambiar tu pin.